src/containers/ActionMenuContainer/ActionMenuContainer.tsx

```diff
@@ -120,124 +124,179 @@ export const ActionMenuContainer: React.FC = () => {
   const { nextShares, currentShares, finishedShares } = estimateUserDebtShares()
 
   return (
-    <WrappedActionMenu
-      onMint={(amount, decimal) => () => {
-        dispatch(actions.mint({ amount: amount.mul(new BN(10 ** 6)).div(new BN(10 ** decimal)) }))
-      }}
-      onBurn={(amount, decimal) => () => {
-        dispatch(
-          actions.burn({
-            amount: amount.mul(new BN(10 ** 6)).div(new BN(10 ** decimal)),
-            tokenAddress: xUSDTokenAddress
-          })
-        )
-      }}
-      onDeposit={(amount, decimal) => () => {
-        dispatch(
-          actions.deposit({
-            amount: amount.mul(
-              new BN(10 ** ((userCollaterals[depositIndex]?.decimals ?? 9) - decimal))
-            ),
-            tokenAddress: userCollaterals[depositIndex]?.programId ?? DEFAULT_PUBLICKEY
-          })
-        )
-      }}
-      onWithdraw={(amount, decimal) => () => {
-        dispatch(
-          actions.withdraw({
-            amount: amount.mul(
-              new BN(10 ** ((userStaked[withdrawIndex]?.decimals ?? 6) - decimal))
-            ),
-            tokenAddress: userStaked[withdrawIndex]?.programId ?? DEFAULT_PUBLICKEY
-          })
-        )
-      }}
-      availableToMint={availableToMint}
-      availableToDeposit={userCollaterals.length ? maxDeposit : new BN(0)}
-      availableToWithdraw={availableToWithdraw}
-      availableToBurn={availableToBurn}
-      mintState={mintState}
-      withdrawState={withdrawState}
-      depositState={depositState}
-      burnState={burnState}
-      xUSDBalance={xUSDTokenAddress.equals(DEFAULT_PUBLICKEY) ? new BN(0) : xUSDBalance.balance}
-      stakingData={{
-        ...userStakingState,
-        stakedUserValue: stakedUserValue,
-        SNYPrice: SNYPrice,
-        allDebtValue: allDebtValue,
-        userMarinadeAmount: userMarinadeAmount,
-        slot: slotState,
-        roundLength: stakingState.roundLength,
-        userDebtShares: userDebtSharesState,
-        rounds: {
-          next: {
-            roundAllPoints: stakingState.nextRound.allPoints,
-            roundPoints: nextShares,
-            roundStartSlot: stakingState.nextRound.start,
-            roundAmount: stakingState.nextRound.amount
+    <>
+      <BurnWarning
+        open={showWarning}
+        burnAmount={{
+          amount: burnAmount,
+          decimal: XUSD_DECIMALS
+        }}
+        onBurn={(amount, decimal) => () => {
+          setShowWarning(false)
+          dispatch(
+            actions.burn({
+              amount: amount.mul(new BN(10 ** 6)).div(new BN(10 ** decimal)),
+              tokenAddress: xUSDTokenAddress
+            })
+          )
+        }}
+        onCancel={() => {
+          setShowWarning(false)
+        }}
+        burnTokenSymbol={userStaked[withdrawIndex]?.symbol ?? 'SNY'}
+        rewards={{
```

Comment:
majority of rewards' props won't be used, leave there only the used ones

Reply:
done